src/newgrf.cpp

```diff
@@ -2173,7 +2172,7 @@ static ChangeInfoResult StationChangeInfo(uint stid, int numinfo, int prop, Byte
 				break;
 
 			case 0x1D: // Station Class name
-				AddStringForMapping(buf.ReadWord(), [statspec](StringID str) { StationClass::Get(statspec->class_index)->name = str; });
+				AddStringForMapping(buf.ReadWord(), [statspec = statspec.get()](StringID str) { StationClass::Get(statspec->class_index)->name = str; });
```

Comment:
I do not know whether there is a reason why you wouldn't want to pass the reference by-reference here as well. Just wondering, as I'm  far from sure that it's actually the better solution. Maybe some C++ guru can give their few bits of currency ;)
```suggestion
				AddStringForMapping(buf.ReadWord(), [&statspec](StringID str) { StationClass::Get(statspec->class_index)->name = str; });
```

Reply:
The reference only exists for the lifetime of this function, whereas the mapping is done later.